### ISSUE TITLE
[Feature] Allow user to add bookmark to widgets

### ISSUE THREAD

**Issue by sumitsum** [appsmithorg/appsmith#2413]
## Summary
1. Allow a user to add a bookmark to any widget so that the user can navigate across the pages easily.

## Motivation
1. This feature will stand out when the user has tens or hundreds of widgets spread across many pages. This feature will make it easy for the user to navigate, especially when widgets are connected in some way and the user needs to navigate back and forth repeatedly.

**Comment by Nikhil-Nandagopal** [comment]
@sumitsum would having search-based navigation to the widget solve this? We're picking that up in the omnibar project
#1641

**Comment by sumitsum** [comment]
@Nikhil-Nandagopal search-based navigation would certainly be an alternative to this feature. However, I see them as complementing each other, rather than a replacement of each other. Ideally, it would be good to have both, but based on the time constraints / current goals that we have, we may decide to delay or drop it.

**Comment by Nikhil-Nandagopal** [comment]
@sumitsum I understand the need. I'm not sure if the solution is right because remembering to bookmark something feels like an added burden as a user. Instead just intelligently telling me the last visited widget / most visited widgets and letting me quickly jump to them via search feels like the right answer.

**Comment by areyabhishek** [comment]
@sumitsum Is this example correct: I have an appsmith app I've built with 10+ pages, 100s of widgets. I should be able to visit the exact form widget I created on page 2 with everything in the exact state that it was. If I bookmark it, then I can find the important widget I was working on, but I haven't touched it for many days.

**Comment by sumitsum** [comment]
@Nikhil-Nandagopal Yes I understand your concern. What I had in mind is something that IntelliJ offers, where in you can bookmark a line but you don't have to remember anything - because they are nameless bookmarks. They don't have any tags associated with them. In order to jump to a bookmark user needs to open a bookmark panel, where all the bookmarks are visible and the user can choose to jump to one of the bookmarks.

**Comment by sumitsum** [comment]
@areyabhishek Yes, it is the kind of use case I had in mind.

**Comment by Nikhil-Nandagopal** [comment]
@sumitsum thanks for the additional information! Could you also add a Gif/video of this feature in IntelliJ?

**Comment by sumitsum** [comment]
@Nikhil-Nandagopal This an older video, but the bookmark feature works exactly the same way in the current version of IntelliJ IDEA as well: https://www.youtube.com/watch?v=wVUU5z1sB3E

**Comment by Nikhil-Nandagopal** [comment]
@sumitsum thanks for the video, the preview is key to this feature!

**Comment by kocharrahul7** [comment]
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 75 |
| Effort (months)  | 0.5 |
